src/Authenticator/TokenAuthenticator.php

```diff
@@ -118,4 +119,14 @@ public function authenticate(ServerRequestInterface $request, ResponseInterface
 
         return new Result($user, Result::SUCCESS);
     }
+
+    /**
+     * No-op method.
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request A request object.
+     * @return void
+     */
+    public function unauthorizedChallenge(ServerRequestInterface $request)
+    {
```

Comment:
I don't think adding the PersistenceInterface is a good idea because a token is not persistent. Its somewhat missleading IMHO.